lib/CL/devices/cuda/pocl-ptx-gen.cc

```diff
@@ -405,49 +409,42 @@ void fixPrintF(llvm::Module *Module) {
   VPrintF->eraseFromParent();
 }
 
-// TODO broken, replaces in whole module not just  1 function
-// Replace all load users of a scalar global variable with new value.
-#if 0
-static void replaceScalarGlobalVar(llvm::Module *Module, const char *Name,
-                                   llvm::Value *NewValue) {
+// Since all helper function are inlined, global variables only need to be replaced by arguments in current function
+static void replaceScalarGlobalVarInFunction(llvm::Module* Module, llvm::Function* Function, const char* Name,
+  llvm::Value* NewValue) {
   auto GlobalVar = Module->getGlobalVariable(Name);
   if (!GlobalVar)
     return;
-
+  
   std::vector<llvm::Value *> Users(GlobalVar->user_begin(),
-                                   GlobalVar->user_end());
+                                  GlobalVar->user_end());
   for (auto *U : Users) {
     auto Load = llvm::dyn_cast<llvm::LoadInst>(U);
     assert(Load && "Use of a scalar global variable is not a load");
-    Load->replaceAllUsesWith(NewValue);
-    Load->eraseFromParent();
+    if (Load->getFunction() == Function) {
+      Load->replaceAllUsesWith(NewValue);
+      Load->eraseFromParent();
+    }
   }
-  GlobalVar->eraseFromParent();
 }
-#endif
-
-// Add an extra kernel argument for the dimensionality.
-void handleGetWorkDim(llvm::Module *Module) {
 
-  llvm::SmallVector<llvm::Function *, 8> FunctionsToErase;
-
-  auto WorkDimVar = Module->getGlobalVariable("_work_dim");
-  if (WorkDimVar == nullptr)
-    return;
+void replaceGlobalVariablesWithArgs(llvm::Module* Module, std::vector<std::string> variableNames, std::vector<llvm::Type*> types) {
```

Comment:
Pls check https://portablecl.org/docs/html/development.html#coding-style. In the C++ (LLVM) style we name variables with capitals: VariableNames etc.  Also there are formatter scripts that help getting the basic format (such as these long lines) cleaned.